astropy/io/fits/util.py

```diff
@@ -612,7 +612,12 @@ def _array_to_file(arr, outfile):
     `ndarray.tofile`.  Otherwise a slower Python implementation is used.
     """
 
-    if isfile(outfile) and not isinstance(outfile, io.BufferedIOBase):
+    try:
+        seekable = outfile.seekable()
+    except AttributeError:
+        seekable = False
+
+    if isfile(outfile) and seekable:
```

Comment:
```suggestion
    if isfile(outfile) and seekable and not (sys.platform.startswith('win')
                                             and isinstance(outfile, io.BufferedIOBase)):
```
Re. my (speculative) comments on those tests that _did_ fail on Windows, maybe you could test if this will let them pass _instead_ of turning the buffering off in general. But in any case I'd wait for @embray's review on this.

Reply:
Ok, I'll go ahead and make those changes so we can test. @dhomeier can you please look at the Windows CI?